### DISCUSSION TITLE
Extended Betza Notation

### DISCUSSION THREAD

**Discussion by musketeerchess** [fairy-stockfish/Fairy-Stockfish#773] [Ideas]
Hi Fabian, i checked the code. adding the Betza Notation is really a very clever idea. This eases a lot customizing unorthodox pieces for many chess variants. If i correctly understood the code, you only added the basic betza notation, but your code doesn't take into account the extended betza notation mentioned here https://www.chessvariants.com/page/MSbetza-notation-extended

Also i saw that when specifying the Leaper atoms, there are 2 duplicates. for example in the original atom table Camel move is C and Zebra move is a (3,2) leap. In your code these two C and Z are doubled with L (3,1) and J (3,2). Is that for a reason?

Is it possible to code with the betza notation in fairy stockfish pieces like this one? 
![68747470733a2f2f692e696d6775722e636f6d2f4c654148456d632e706e67](https://github.com/fairy-stockfish/Fairy-Stockfish/assets/34623005/66bcc9cd-43e1-45f2-b6e3-bbfc48bf545c)
![68747470733a2f2f692e696d6775722e636f6d2f6f524a387347462e706e67](https://github.com/fairy-stockfish/Fairy-Stockfish/assets/34623005/26ccde18-9da6-44b8-94d1-b7cef8106e80)

Thanks Fabian

**Comment by ianfab** [comment]
In Fairy-SF Betza notation is essentially just a means for communicating the piece moves, it does not really define what is supported/possible in the engine, since that needs to be implemented separately. Therefore, only a limited subset of Betza notation is supported, as specified in the documentation, see https://github.com/fairy-stockfish/Fairy-Stockfish/blob/f3c982bdb5a1882c45d9fab71b16f72de5191f19/src/variants.ini#L100-L107

**Comment by musketeerchess** [comment]
Hi Fabian, Thank you for your kind answer.
Are you willing to extend Betza notation? What are your current top
priorities? Hope you'll find sometime to help us concerning Musketeer
Chess. The project advanced very well with various people trying to find a
solution. But there are still a few key steps to improve the quality of
computation.

Le mer. 3 avr. 2024 à 20:44, Fabian Fichter ***@***.***> a
écrit :

**Comment by seocwen** [comment]
I've recently been playing a lot of Ouroboros King on Steam, which seems to use a custom, kind of bad engine for fairy chess. It has a lot of pretty cool pieces that are fun to play with, like the manticore, and griffon, notated t[WB] and t[FR] respectively. Implementation-wise, I wonder if something like that could be coded as moving twice. I think there's at least one more piece using the t[??] notation in the historical variant Grant Acedrex (which is usually played 12x12, but a 12x10 variant would also be fun.
